src/utils/create-utils.ts

```diff
@@ -151,3 +160,73 @@ export const createComponentDetectionQuery = async (
 
   return cdq.status.componentDetected;
 };
+
+/**
+ * Create SPIAccessTokenBinding CR
+ *
+ * @param url the URL of the git repository
+ * @param namespace namespace to create the binding
+ * @returns Returns created SPIAccessTokenBinding resource
+ */
+export const createAccessTokenBinding = async (url: string, namespace: string) => {
+  const id = uid();
+  const requestData = {
+    apiVersion: `${SPIAccessTokenBindingModel.apiGroup}/${SPIAccessTokenBindingModel.apiVersion}`,
+    kind: SPIAccessTokenBindingModel.kind,
+    metadata: {
+      name: `hac-dev-import-${id}`,
+      namespace,
+      labels: {
+        [SPI_AUTH_REPO_LABEL]: 'hac-dev',
+      },
+    },
+    spec: {
+      repoUrl: url,
+      permissions: {
+        required: [
+          { type: 'r', area: 'repository' },
+          { type: 'w', area: 'repository' },
+        ],
+      },
+      secret: {
+        name: `hac-dev-token-${id}`,
+        type: 'kubernetes.io/basic-auth',
+      },
+    },
+  };
+
+  const binding: SPIAccessTokenBindingKind = await k8sCreateResource({
+    model: SPIAccessTokenBindingModel,
+    data: requestData,
+  });
+
+  return binding;
+};
+
+/**
+ * Create SPIAccessTokenBinding if a binding for the same Git URL does not already exist.
+ * Else return the existing SPIAccessTokenBinding.
+ *
+ * @param url the URL of the git repository
+ * @param namespace namespace to create the binding
+ * @returns Returns the SPIAccessTokenBinding resource
+ */
+export const initiateAccessTokenBinding = async (url: string, namespace: string) => {
+  const bindings: SPIAccessTokenBindingKind[] = await k8sListResource({
+    model: SPIAccessTokenBindingModel,
+    queryParams: {
+      ns: namespace,
+      labelSelector: {
+        matchLabels: {
+          [SPI_AUTH_REPO_LABEL]: 'hac-dev',
+        },
+      },
+    },
+  });
+  const binding = bindings.find((b) => b.spec.repoUrl === url);
```

Comment:
reusing the binding is preferred

Reply:
Added check for comparing permissions.

Reply:
@christianvogt since it's preferred to reuse bindings, should we not delete any binding when moving away from the page? Or maybe delete only the ones created by UI?

Reply:
That's a good question.
I hate to leave around resources when the flow wasn't completed however it probably doesn't hurt since we will be doing a lookup to reuse.
Let's leave them for now and we can revisit this later.

Reply:
Removed deletion hook.